api/logs/v1/http.go

```diff
@@ -131,6 +131,10 @@ func NewHandler(read, tail, write *url.URL, upstreamCA []byte, opts ...HandlerOp
 				t.TLSClientConfig.RootCAs.AppendCertsFromPEM(upstreamCA)
 			}
 
+			if upstreamCert != nil {
+				t.TLSClientConfig.Certificates = append(t.TLSClientConfig.Certificates, *upstreamCert)
```

Comment:
If `upstreamCA` is empty then `TLSClientConfig` is still `nil` here. In that case it will panic here.